File: app/src/main/java/com/example/attenda_attempt3/StudentInformationFormActivity.java
package com.example.attenda_attempt3;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.HashMap;
import java.util.Map;

public class StudentInformationFormActivity extends AppCompatActivity {

    Button btnNextStep;
    public static EditText etFirstAndLastName;
    EditText etSchoolID;
    FirebaseFirestore db = FirebaseFirestore.getInstance();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_student_information_form);

        btnNextStep = findViewById(R.id.btnUpdateInfo);
        etFirstAndLastName = findViewById(R.id.etFullName);
        etSchoolID = findViewById(R.id.etSchoolID);

        btnNextStep.setOnClickListener(view -> {
            String fullName = etFirstAndLastName.getText().toString();
            String schoolID = etSchoolID.getText().toString();

            String uid = FirebaseAuth.getInstance().getCurrentUser().getUid();
            String uid2 = FirebaseAuth.getInstance().getCurrentUser().getUid();

            Map<String, Object> data = new HashMap<>();
            data.put("userID", uid2);

            db.collection("users").document(uid).set(data)
                    .addOnSuccessListener(unused -> {
                        Map<String, Object> userdata = new HashMap<>();
                        userdata.put("studentName", fullName);
                        userdata.put("schoolID", schoolID);
                        db.collection("users").document(uid).update(userdata)
                                .addOnSuccessListener(unused1 -> {
                                    Toast.makeText(getApplicationContext(), "Data Saved", Toast.LENGTH_SHORT).show();
                                    Intent intent = new Intent(StudentInformationFormActivity.this, StudentScheduleSelectActivity.class);
                                    startActivity(intent);
                                    finish();
                                })
                                .addOnFailureListener(e -> Toast.makeText(getApplicationContext(), "Error Occurred, Could Not Switch Activities", Toast.LENGTH_SHORT).show());
                    })
                    .addOnFailureListener(e -> Toast.makeText(getApplicationContext(), "Error Occurred, Data Not Saved", Toast.LENGTH_SHORT).show());
        });

    }
}

File: app/src/main/java/com/example/attenda_attempt3/StudentInformationFormActivity.kt
package com.example.attenda_attempt3

import android.content.Intent
import android.os.Bundle
import android.view.WindowManager
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInClient
import com.google.android.gms.auth.api.signin.GoogleSignInOptions
import com.google.firebase.auth.FirebaseAuth

class StudentInformationFormActivity : AppCompatActivity() {
    lateinit var mGoogleSignInClient: GoogleSignInClient
    private val auth by lazy {
        FirebaseAuth.getInstance()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_student_information_form)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )

        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
            .requestIdToken(getString(R.string.default_web_client_id))
            .requestEmail()
            .build()
        mGoogleSignInClient = GoogleSignIn.getClient(this, gso)
        val btnSignOut: Button = findViewById(R.id.btnSignOut)
        btnSignOut.setOnClickListener {
            mGoogleSignInClient.signOut().addOnCompleteListener {
                val intent = Intent(this, LoginActivityKotlin::class.java)
                startActivity(intent)
                finish()
            }
        }
    }
}